docs/concepts/task-queues.md

```diff
@@ -4,25 +4,70 @@ title: Task Queues
+## What is a Task Queue?
 
-Instead of calling the Worker directly, an intermediate queue is used. So the service adds an Activity Task to this queue and a Worker receives the Task using a long poll request.
+There are a few different perspectives from which we can talk about Task Queues.
 
-Temporal calls this queue used to dispatch Activity Tasks an Activity Task Queue.
+import SharedTaskQueuesBasic from '../shared/task-queues-basic.md'
 
-Similarly, when a Workflow needs to handle an external event, a Command Task is created.
-A Command Task Queue is used to deliver it to the Worker.
+<SharedTaskQueuesBasic
+workflowLink="/docs/concepts/workflows"
+workerLink="/docs/concepts/workers"
+/>
 
-While Temporal Task Queues are queues, they have some differences from commonly used queuing technologies. The main one is that they do not require explicit registration and are created on demand. The number of Task Queues is not limited. A common use case is to have a Task Queue per Worker process and use it to deliver Activity tasks to the process. Another use case is to have a Task Queue per pool of Workers.
+From the perspective of a developer, using the SDK, Task Queues are one of the means by which you associate a Worker with a Workflow and/or Activity.
+In this case, you can learn about how to implement Task Queues within the context of the language you are writing your application in:
 
-There are multiple advantages of using a Task Queue to deliver Tasks instead of invoking an Activity Worker through a synchronous RPC:
+- [Go](/docs/go/task-queues)
+- [Java](/docs/java/task-queues)
+- [PHP](/docs/php/task-queues)
 
-- Worker doesn't need to have any open ports, which is more secure.
-- Worker doesn't need to advertise itself through DNS or any other network discovery mechanism.
-- When all Workers are down, messages are persisted in a Task Queue waiting for the Workers to recover.
+From the perspective of the system's design and how everything works under the hood, things get a little bit more complex.
+
+We intend to explain this in more detail in future system design and architecture docs.
+
+## Why Task Queues?
+
+Temporal Task Queues are a little bit different from commonly used queuing technologies.
+The main difference being that they do not require explicit registration and are created on demand.
+Task Queues are very lightweight and there is no limit to the total number of Task Queues that the system can handle.
+
+There are multiple advantages of using a Task Queue to deliver Tasks to a Worker, instead of invoking actions via a synchronous RPC.
+
+- Workers do not need to have any open ports, which is more secure.
+- Workers do not need to advertise themselves through DNS or any other network discovery mechanism.
+- When all Workers are down, messages simply persist in a Task Queue, waiting for the Workers to recover.
 - A Worker polls for a message only when it has spare capacity, so it never gets overloaded.
```

Comment:
I realize this is the one line that didn't change, lol, but "never" might be too strong of working here. Users could still do weird things in their code causing workers to slow down, even with only one task, for example.

```suggestion
- A Worker polls for a message only when it has spare capacity, avoiding overloading itself.
```

Reply:
SG!